DigitalLearningSolutions.Data.Tests/DataServices/LearningLogItemsDataServiceTests.cs

```diff
@@ -0,0 +1,131 @@
+﻿namespace DigitalLearningSolutions.Data.Tests.DataServices
+{
+    using System;
+    using System.Transactions;
+    using DigitalLearningSolutions.Data.DataServices;
+    using DigitalLearningSolutions.Data.Tests.TestHelpers;
+    using FluentAssertions;
+    using FluentAssertions.Execution;
+    using NUnit.Framework;
+
+    public class LearningLogItemsDataServiceTests
+    {
+        private ILearningLogItemsDataService service = null!;
+        private LearningLogItemsTestHelper learningLogItemsTestHelper = null!;
+        private CompetencyLearningResourcesTestHelper competencyLearningResourcesTestHelper = null!;
+
+        [SetUp]
+        public void Setup()
+        {
+            var connection = ServiceTestHelper.GetDatabaseConnection();
+            service = new LearningLogItemsDataService(connection);
+
+            learningLogItemsTestHelper = new LearningLogItemsTestHelper(connection);
+            competencyLearningResourcesTestHelper = new CompetencyLearningResourcesTestHelper(connection);
+        }
+
+        [Test]
+        public void InsertLearningLogItem_inserts_expected_record()
+        {
+            using var transaction = new TransactionScope();
+            try
+            {
+                // Given
+                const int competencyLearningResourceId = 1;
+                const int delegateId = 2;
+                const string resourceName = "Activity";
+                const string resourceLink = "www.test.com";
+                var addedDate = new DateTime(2021, 11, 1);
+                competencyLearningResourcesTestHelper.InsertCompetencyLearningResource(
+                    1,
+                    competencyLearningResourceId,
+                    1,
+                    7
+                );
+
+                // When
+                service.InsertLearningLogItem(
+                    delegateId,
+                    addedDate,
+                    resourceName,
+                    resourceLink,
+                    competencyLearningResourceId
+                );
+                var result = learningLogItemsTestHelper.SelectLearningLogItemWithResourceLink(resourceLink);
+
+                // Then
+                using (new AssertionScope())
+                {
+                    result.Should().NotBeNull();
+                    result!.DelegateId.Should().Be(delegateId);
+                    result.AddedDate.Should().Be(addedDate);
+                    result.CompetencyLearningResourceId.Should().Be(competencyLearningResourceId);
+                    result.ResourceLink.Should().Be(resourceLink);
+                    result.ResourceName.Should().Be(resourceName);

```

Comment:
Do you think it would be worth checking the other columns are null, just to make sure illogical additions are not made (e.g. CompletedDate not being null etc.)?

Reply:
Yeah probably.
If I'm adding a new LearningLogItems class for the below, I guess it needs to have properties for all the columns in the database, and we can then check the rest are null with a full select